ember_debug_test_app/tests/helpers/setup-ember-debug-test.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable ember/no-runloop */
+import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import EmberRouter from '@ember/routing/router';
+import {
+  getApplication,
+  setApplication,
+  setupApplicationContext,
+  setupContext,
+  teardownContext,
+} from '@ember/test-helpers';
+import { run } from '@ember/runloop';
+import BasicAdapter from 'test-app/services/adapters/basic';
+import config from 'test-app/config/environment';
+import EmberDebug from 'ember-debug/main';
+import { hbs } from 'ember-cli-htmlbars';
+import Port from 'ember-debug/port';
+
+export default function setupEmberDebugTest(hooks, options = {}) {
+  let app, originalApp, originalPort, originalIgnoreDeprecations;
+
+  hooks.beforeEach(async function () {
+    originalPort = EmberDebug.Port;
+    originalApp = getApplication();
+    originalIgnoreDeprecations = EmberDebug.IGNORE_DEPRECATIONS;
+
+    app = Application.create({
+      ...config.APP,
+      modulePrefix: config.modulePrefix,
+      Resolver,
+    });
+    setApplication(app);
+
+    await setupContext(this);
+    await setupApplicationContext(this);
+
+    // eslint-disable-next-line ember/no-classic-classes
+    const Router = EmberRouter.extend({
+      location: 'none',
+    });
+
+    if (options.routes) {
+      Router.map(options.routes);
+    } else {
+      Router.map(function () {
+        this.route('simple');
+      });
+
+      this.owner.register('template:simple', hbs`Simple template`);
+    }
+
+    this.owner.register('router:main', Router);
+    this.owner.register('service:adapter', BasicAdapter);
```

Comment:
Hey @patricklx, I am very unfamiliar with the inspector code and have a question about this. The way I understand this code is "The app I want to test is not the inspector UI app, but a new simple app in which the ember_debug runs", so we are creating this simple app in the test.

If you move the ember_debug tests in their own Ember app package, is this code still useful? Shouldn't the `ember_debug_test_app` be what is described by the present hook? 

Please let me know if I misunderstand what you're doing 🙏

Reply:
Yes, I understand the ultimate goal of your PR and why it's fixing the matrix 👍  My question was more about a specific part of the code to achieve this, but maybe the piece I am missing is about the exact mechanics of `ember-try`, never mind.

Reply:
Ah, sorry, it's just a coincidence that my comment stops on the `BasicAdapter` line (I wanted to highlight the whole app creation block), but anyway no worries. 

I see that `helpers-setup-ember-debug-test.js` has been "created" rather than "moved", but I believe this helper is specific to `ember_debug` tests and can be removed from the main tests folder? I also can't find any reference to `trigger-port` function, I wonder if that's just some legacy artefact that could be removed 🤔